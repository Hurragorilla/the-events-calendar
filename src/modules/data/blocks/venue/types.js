/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_VENUE = `${ PREFIX_EVENTS_STORE }/SET_VENUE`;
export const SET_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP`;
export const SET_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP_LINK`;

export const ADD_BLOCK_VENUE = `${ PREFIX_EVENTS_STORE }/ADD_BLOCK_VENUE`;
export const REMOVE_BLOCK_VENUE = `${ PREFIX_EVENTS_STORE }/REMOVE_BLOCK_VENUE`;
