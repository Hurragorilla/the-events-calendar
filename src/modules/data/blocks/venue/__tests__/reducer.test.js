/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/venue';
import reducer from '@moderntribe/events/data/blocks/venue/reducer';

jest.mock( '@moderntribe/common/utils/globals', () => ( {
	dateSettings: () => ( {} ),
	editorDefaults: () => ( {
		venue: 0,
		venueCountry: '',
		venueState: '',
		venueProvince: '',
	} ),
	list: () => ( {
		countries: {},
		us_states: {},
	} ),
	mapsAPI: () => ( {
		embed: true,
	} ),
} ) );

describe( '[STORE] - Venue reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toMatchSnapshot();
	} );

	it( 'Should set the venue', () => {
		expect( reducer( {}, actions.setVenue( 99 ) ) ).toMatchSnapshot();
	} );

	it( 'Should remove the venue', () => {
		expect( reducer( {}, actions.removeVenue() ) ).toMatchSnapshot();
	} );
} );
