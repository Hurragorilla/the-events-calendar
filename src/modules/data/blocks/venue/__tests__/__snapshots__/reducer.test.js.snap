// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Venue reducer Should remove the venue 1`] = `
Object {
  "blocks": Object {
    "allIds": Array [],
    "byId": Object {},
    "core": Object {},
  },
}
`;

exports[`[STORE] - Venue reducer Should return the default state 1`] = `
Object {
  "blocks": Object {
    "allIds": Array [],
    "byId": Object {},
    "core": Object {},
  },
}
`;

exports[`[STORE] - Venue reducer Should set the venue 1`] = `
Object {
  "blocks": Object {
    "allIds": Array [],
    "byId": Object {},
    "core": Object {},
  },
}
`;
